aten/src/ATen/native/Pow.cpp

```diff
@@ -22,9 +22,10 @@ Tensor& pow_out(Tensor& result, const Tensor& base, Scalar exp) {
   TORCH_CHECK(!(isIntegralType(base.scalar_type(), true) &&
               exp.isIntegral(true) && exp.toLong() < 0),
               "Integers to negative integer powers are not allowed.");
-  if (exp.toDouble() == 0.0) {
+  // Avoid runtime error when typecasting
+  if (!exp.isComplex() && (exp.toDouble() == 0.0)) {
```

Comment:
on more thought, this should probably just be `exp == 0` since we also want to return `result.resize_as_(base).fill_(1);` for exp = 0 + 0j

Reply:
I am not sure why we are converting the exponent to double before the comparison though.

cc. @nairbv